Fix baCloseAll with nested elements within a baOpen/etc. element

cjh79 opened the pull request:
Fix issue where, when using baCloseAll, clicking on a nested element
within a baOpen/baClose/baToggle element would not open the thing it
was supposed to open. When checking for a click on such an element,
check the element’s ancestry as well.

Fixes issue #85 

Tested on browsers:
Firefox (OSX, latest)
Chrome (OSX, latest)
IE11
Safari 10
Chrome, Firefox (Android)
Chrome, Safari (iOS)

Commit:
Fix baCloseAll with nested elements

Fix issue where, when using baCloseAll, clicking on a nested element
within a baOpen/baClose/baToggle element would not open the thing it
was supposed to open. When checking for a click on such an element,
check the element’s ancestry as well.

soumak77 commented on js/angular/components/common/common.js:
```diff
@@ -203,7 +203,7 @@
         // check if clicked target is designated to open/close another component
         avoid = ['ba-toggle', 'ba-hard-toggle', 'ba-open', 'ba-close'].filter(function(e){
-          return e in tar.attributes;
+          return tar.closest('*[' + e + ']') !== null;
```
`closest` is not supported by [angular.element](https://docs.angularjs.org/api/ng/function/angular.element) and is [not a fully supported](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest#Browser_compatibility) across all browsers.

You'll need to manually search up the tree using `parent`.  See the following examples inside this file: 
- https://github.com/base-apps/angular-base-apps/blob/master/js/angular/components/common/common.js#L269-L281
- https://github.com/base-apps/angular-base-apps/blob/master/js/angular/components/common/common.js#L33-L43

cjh79 replied:
Ah - jqLite, right. Will update the fix today. Thanks.

Commit:
jqlite-friendly fix for issue #85

Fix issue where nested elements within a ba-open (etc) element would
not open the thing that was supposed to be opened.

cjh79 commented:
I have committed a new fix which does not use closest(). I have tested it on the same set of browsers as mentioned in my original message on this PR.

I see that the Travis CI job failed. I'm not sure why, it says there was an error while generating the build script but nothing else. Please let me know if there is something I need to do to address this.
